trade-finance-manager-ui/server/controllers/case/index.test.js

```diff
@@ -7,6 +7,11 @@ import mapAssignToSelectOptions from '../../helpers/map-assign-to-select-options
 
 const res = mockRes();
 
+jest.mock('@ukef/dtfs2-common', () => ({
```

Comment:
I'm not super sure what to do in this test file - the existing tests just check very broad high level things such as 'when the deal exists the deal template is rendered' and doesn't check specific sub cases of values. 

I could break that pattern and add in conditional testing for the `showCancelDealButton` e.g. tests of the form: 

When the TFM Deal cancellation FF is enabled -> 
- If the deal type is AIN then showCancelDealButton should be true
-  If the deal type is MIN then showCancelDealButton should be true
-  If the deal type is MIA then showCancelDealButton should be false

When the TFM Deal cancellation FF is disabled -> 
- If the deal type is AIN then showCancelDealButton should be false
-  If the deal type is MIN then showCancelDealButton should be false
-  If the deal type is MIA then showCancelDealButton should be false

But I feel this logic is tested in unit tests, and ultimately will be tested in E2E tests too. 
It still needs some mock of `isTfmDealCancellationFeatureFlagEnabled` hence this at the top of the file. 

What do you think is best here?

Reply:
Is this an api test or a unit test? It seems like it wants to be a unit test (as per its name), so I think the only thing we care for is what this file sees, and no further -- ie it does not care about implimentation of ` dealCancellationEnabled(deal.dealSnapshot.submissionType)`, and therefore I dont think it cares about the feature flag at all.

If I were to approach this, I'd look at mocking `dealCancellationEnabled(deal.dealSnapshot.submissionType)`, asserting that `dealCancellationEnabled` is called with the expected value, and any value it returns (which we mock in the test) is passed on... How does that sound?

So we could have `it('should check whether deal cancellation is enabled'` and a `it('should render the template with the deal cancellation value')` or something similar -- the namings slightly off there!

Reply:
Great point, I've now added unit tests to this effect 👍